services/121-service/src/registration-debit-cards/registration-debit-cards.controller.ts

```diff
@@ -0,0 +1,202 @@
+import {
+  Controller,
+  Get,
+  HttpCode,
+  HttpException,
+  HttpStatus,
+  Param,
+  ParseBoolPipe,
+  ParseIntPipe,
+  Patch,
+  Post,
+  Query,
+  Req,
+  UseGuards,
+} from '@nestjs/common';
+import {
+  ApiOperation,
+  ApiParam,
+  ApiQuery,
+  ApiResponse,
+  ApiTags,
+} from '@nestjs/swagger';
+
+import { AuthenticatedUser } from '@121-service/src/guards/authenticated-user.decorator';
+import { AuthenticatedUserGuard } from '@121-service/src/guards/authenticated-user.guard';
+import { IntersolveVisaWalletDto } from '@121-service/src/payments/fsp-integration/intersolve-visa/dtos/internal/intersolve-visa-wallet.dto';
+import { RegistrationDebitCardsService } from '@121-service/src/registration-debit-cards/registration-debit-cards.service';
+import { PermissionEnum } from '@121-service/src/user/enum/permission.enum';
+import { UserService } from '@121-service/src/user/user.service';
+
+@UseGuards(AuthenticatedUserGuard)
+@Controller()
+export class RegistrationDebitCardsController {
+  public constructor(
+    private readonly userService: UserService,
+    private readonly registrationDebitCardsService: RegistrationDebitCardsService,
+  ) {}
+
+  // Re-issue card: this is placed in RegistrationDebitCardsController because it also sends messages and searches by referenceId
+  @ApiTags('fsps/intersolve-visa')
+  @AuthenticatedUser({ permissions: [PermissionEnum.FspDebitCardCREATE] })
+  @ApiOperation({
+    summary: '[SCOPED] Re-issue card: replace existing card with a new card.',
+  })
+  @ApiParam({ name: 'programId', required: true, type: 'integer' })
+  @ApiParam({ name: 'referenceId', required: true, type: 'string' })
+  @ApiResponse({
+    status: HttpStatus.OK,
+    description:
+      'Card replaced - NOTE: this endpoint is scoped, depending on program configuration it only returns/modifies data the logged in user has access to.',
+  })
+  @Post(
+    'programs/:programId/registrations/:referenceId/fsps/intersolve-visa/wallet/cards',
+  )
+  @HttpCode(HttpStatus.NO_CONTENT)
+  public async reissueCardAndSendMessage(
+    @Param('programId', ParseIntPipe) programId: number,
+    @Param('referenceId') referenceId: string,
+    @Req() req,
+  ): Promise<void> {
+    const userId = req.user.id;
+    await this.registrationDebitCardsService.reissueCardAndSendMessage(
+      referenceId,
+      programId,
+      userId,
+    );
+  }
+
+  @ApiTags('fsps/intersolve-visa')
```

Comment:
if the ApiTags are the same for every endpoint in a controller, you can just put it on top once above @Controller.